### ISSUE TITLE
Kabanero operator fails to start with an "exe format error" when the operator image built on MAC

### ISSUE THREAD

**Issue by mezarin** [kabanero-io/kabanero-operator#233]
The kabanero operator pod shows the following line in it's log:
standard_init_linux.go:211: exec user process caused "exec format error"

It appears that when the GO code is built, the wrong OS ENVs are being used so that the code is built incorrectly for the environment where it runs.
 
If i update this line: 

CGO_ENABLED=0 go build -o build/_output/bin/kabanero-operator -gcflags "all=-trimpath=$(GOPATH)" -asmflags "all=-trimpath=$(GOPATH)" -ldflags "-X main.GitTag=$(TRAVIS_TAG) -X main.GitCommit=$(TRAVIS_COMMIT) -X main.GitRepoSlug=$(TRAVIS_REPO_SLUG) -X main.BuildDate=`date -u +%Y%m%d.%H%M%S`" github.com/kabanero-io/kabanero-operator/cmd/manager

To this:

CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o build/_output/bin/kabanero-operator -gcflags "all=-trimpath=$(GOPATH)" -asmflags "all=-trimpath=$(GOPATH)" -ldflags "-X main.GitTag=$(TRAVIS_TAG) -X main.GitCommit=$(TRAVIS_COMMIT) -X main.GitRepoSlug=$(TRAVIS_REPO_SLUG) -X main.BuildDate=`date -u +%Y%m%d.%H%M%S`" github.com/kabanero-io/kabanero-operator/cmd/manager

The issue goes away.

We should either revert to using the operator-sdk command or handle the cases the operator-sdk handles (i.e -x86_64-linux-gnu vs -ppc64le-linux-gnu, etc)

**Comment by kaczyns** [comment]
In #226 I changed the build to stop using `operator-sdk build` and instead use the two separate commands that it calls:  `go build` and `docker build`.  When I converted, I neglected this code:
https://github.com/operator-framework/operator-sdk/blob/33b3bfe10176f8647f5354516fff29dea42b6342/cmd/operator-sdk/build/cmd.go#L88

This sets GOOS and GOARCH to linux and amd64 respectively.  We should be doing the same.  Someday when we do multi-arch builds (for powerpc and zlinux) we'll need to change this.
